File: Python/longest_uncommon_subsequence_1.py
# https://leetcode.com/problems/longest-uncommon-subsequence-i/description/
from utils.examination import Examiner, Testcase


class Solution:
    def findLUSlength(self, a: str, b: str) -> int:
        if a == b:
            return -1
        return max(len(a), len(b))

def examine()->None:
    Examiner(Solution(), 'findLUSlength').run((
        Testcase(('aba', 'cdc'), 3),
        Testcase(('aaa', 'bbb'), 3),
        Testcase(('aac', 'bbc'), 3),
        Testcase(('aca', 'bcb'), 3),
        Testcase(('aaa', 'a'), 3),
        Testcase(('aaa', 'aaa'), -1),
    ))

File: TypeScript/solutions/longest-uncommon-subsequence-1.ts
// https://leetcode.com/problems/longest-uncommon-subsequence-i/description/
import { Examiner, Testcase } from '../utils';


function findLUSlength(a: string, b: string): number {
    if (a === b) return -1;
    return Math.max(a.length, b.length);
}

export function examine(): void {
    new Examiner(findLUSlength).run([
        new Testcase(['aba', 'cdc'], 3),
        new Testcase(['aaa', 'bbb'], 3),
        new Testcase(['aac', 'bbc'], 3),
        new Testcase(['aca', 'bcb'], 3),
        new Testcase(['aaa', 'a'], 3),
        new Testcase(['aaa', 'aaa'], -1),
    ]);
}
